blog/v8-memory-cage.md

```diff
@@ -0,0 +1,87 @@
+---
+title: Electron and the V8 Memory Cage
+date: 2022-06-30T00:00:00.000Z
+authors:
+  - name: nornagon
+    url: 'https://github.com/nornagon'
+    image_url: 'https://github.com/nornagon.png?size=96'
+slug: v8-memory-cage
+---
+
+Electron 20 and later will have the V8 Memory Cage enabled, with implications for some native modules.
+
+---
+
+In Electron 20, we will be enabling [V8 sandboxed pointers](https://docs.google.com/document/d/1HSap8-J3HcrZvT7-5NsbYWcjfc0BVoops5TDHZNsnko/edit) in Electron, following Chrome's [decision to do the same in Chrome 103](https://chromiumdash.appspot.com/commit/9a6a76bf13d3ca1c6788de193afc5513919dd0ed). This has some implications for native modules. Also, we previously enabled a related technology, [pointer compression](https://v8.dev/blog/pointer-compression), in Electron 14. We didn't talk about it much then, but pointer compression has implications for the maximum V8 heap size.
+
+These two technologies, when enabled, are significantly beneficial for security, performance and memory usage. However, there are some downsides to enabling them, too.
+
+The main downside of enabling sandboxed pointers is that **ArrayBuffers which point to external ("off-heap") memory are no longer allowed**. This means that native modules which rely on this functionality in V8 will need to be refactored to continue working in Electron 20 and later.
+
+The main downside of enabling pointer compression is that **the V8 heap is limited to a maximum size of 4GB**. The exact details of this are a little complicated-for example, ArrayBuffers are counted separately from the rest of the V8 heap, but have their [own limits](https://bugs.chromium.org/p/chromium/issues/detail?id=1243314).
```

Comment:
```suggestion
The main downside of enabling pointer compression is that **the V8 heap is limited to a maximum size of 4GB**. The exact details of this are a little complicated — for example, ArrayBuffers are counted separately from the rest of the V8 heap, but have their [own limits](https://bugs.chromium.org/p/chromium/issues/detail?id=1243314).
```

Reply:
Don't want to get too nitpicky but I see the hyphen character in the formatted preview too: https://github.com/electron/electronjs.org-new/blob/cb653563ca41809697e50c08c7c1f91f39e1a344/blog/v8-memory-cage.md

<img width="634" alt="image" src="https://user-images.githubusercontent.com/16010076/180078879-3475dd1f-cef7-4091-9457-9b2311891cde.png">

vs

<img width="642" alt="image" src="https://user-images.githubusercontent.com/16010076/180079149-be915ca0-ed32-485d-827a-d402663ed94b.png">


Didn't know that tidbit about no spaces though!

Reply:
ugh you're right, these must have gotten lost when i copy/pasted from medium. thanks.